commit/merkleroot/rmn/types/config.go

```diff
@@ -43,8 +43,8 @@ type RemoteConfig struct {
 }
 
 func (r RemoteConfig) IsEmpty() bool {
-	return len(r.ContractAddress) == 0 &&
-		r.ConfigDigest == (cciptypes.Bytes32{}) &&
+	// NOTE: contract address will always be present, since the code auto populates it
```

Comment:
If it's always present, why remove the length check?

Reply:
Because we want this check to return `IsEmpty` in order to skip the rest of the config validation (or merkle root observations fail).

The `RMNRemote` contract is bound so that Solana can use `GetCursedSubjects`, but we don't actually want to enable the full config here, which would imply RMNHome support. We blank out the `Version` value etc to return an empty config, but `ContractAddress` is the one variable we can't control: https://github.com/smartcontractkit/chainlink-ccip/blob/721ad77875e556708f1ecb76a4ac4b02e19eaaa1/pkg/reader/ccip.go#L1007